### PR TITLE
fix(robocasa): pin VLA support resources and document installation checks

### PR DESCRIPTION
**Pull request by ye1457**
## Summary

Integrate the merged RoboCasa external/read-only assets fixes and RLDX support-resource loading with consolidated English and Chinese setup guides. Based on main `849143b`; existing Franka, tutorial and RoboTwin changes are preserved.

- Ordinary installation and Target50 both use the maintained **`rpent` branches** of RoboCasa, Robosuite and RLDX. The manifest records `branch: "rpent"`; no second fixed-SHA installation or override file is required. Users record resolved commits and installed versions with `uv pip freeze`, and check compatibility with `uv pip check`.
- Keep `rpent-robocasa365`, the distribution name actually declared by RoboCasa's `rpent` branch. Renaming the requirement to `rlinf-robocasa365` without a corresponding upstream metadata change would be inconsistent. Upstream naming remains open for maintainer coordination.
- HF checkpoint, backbone-support and task-memory revisions stay fixed. Moving Git branches are not represented as immutable source snapshots.
- The RoboCasa VLA worker automatically selects the documented backbone-support snapshot; no new RPent/server revision flag or cache ref mutation. Support metadata does not replace finetuned weights.
- Keep necessary compatibility constraints, but no exact accelerator pins or machine-specific wheel. Users select a compatible Torch/torchvision/CUDA combination.
- Remove the duplicate robot-directory README and consolidate resource, recovery, cache and memory instructions in the official bilingual usage pages.
- Reuse the existing RLDX component test and four representative environment/camera smoke cases. No planner, shared RPC or CLI-framework changes.

## Dependency Sources

| Dependency | Source branch | Merged fixes |
| --- | --- | --- |
| RoboCasa | `RLinf/robocasa@rpent` | RLinf/robocasa#3 and #4 |
| Robosuite | `RLinf/robosuite@rpent` | Omron navview support |
| RLDX | `RLinf/RLDX-1@rpent` | RLinf/RLDX-1#2 |

The actual distribution name is `rpent-robocasa365`. Do not co-install another distribution providing the same `robocasa` import package. Backbone support remains `RLWRLD/RLDX-1-VLM@4b9f870d1287e0d38d7eb1445e6d8c60afe66dd7` (non-weight support files).

## Verification (2026-09-16)

- 588 CPU tests passed, 3 skipped; one existing Starlette deprecation warning.
- Pre-commit, strict full English and Chinese Sphinx builds and diff checks passed.
- Contracts check the three manifest branches against the actual extra requirements, retain HF snapshots, and reject the removed duplicate source-install/override instructions.
- Fresh Python 3.10.12 virtual environment: the documented editable `.[robocasa]` installation with the Target50 constraints completed, and `uv pip check` passed for all 261 installed packages. Verified Torch 2.7.0+cu126 / torchvision 0.22.0+cu126, package imports, `rpent --help`, RoboCasa-specific help, and the asset installer's `--overwrite` option. RLDX policy and RoboCasa VLA imports also passed without loading a model. A second installation completed successfully.
- Installed Git metadata confirms the requested revision is `rpent` for all three dependencies: RoboCasa `2692d8fc5fd86708a1b2028dcbce892ec418a9e7`, Robosuite `97cfbde4b68d8ec43dad20cf4747297866a6ca2e`, and RLDX `ebcfd13df5177e4b3e574bdf5a3b427c7c4a1e8a`. These are observed installation commits, not new manifest pins.
- Installation used a public PyPI mirror, official PyTorch CUDA wheels and verified public-source caches. All 2,078 files in the official RLDX source archive were matched against the Git commit's blob hashes before caching. This is fresh-environment verification, not an empty-cache/offline-install claim; no existing `site-packages` or migration patches were copied.
- No formal experiment or shared user environment was changed. Task matrix, memory scope, scoring and runtime behavior are unchanged by the branch-policy review fix.

## Remaining Acceptance

Open for code review, not a full reproduction certification. Fresh environment installation is verified; full assets/checkpoint download, final GPU component validation and the six-cell accuracy gate remain to be completed for this candidate. Upstream package naming needs maintainer coordination.

Task-only memory remains `551fc3157b3e56b40a3d3a3b4c7ff81721ebe89b`: 111 files with optional Markdown and no global memory. Checkpoint revision, ordinary `--seed`, budgets, task-weighted metrics and Target50 `max_chunks=40` versus ordinary 70 are unchanged. No runner, sandbox or supervisor is introduced. The existing wheel omission of top-level `robots/**` is outside this PR; use the documented editable checkout.

### CODE REVIEW

File: robots/robocasa/README.md [hunk 1]
```diff
@@ -57,6 +57,13 @@ robocasa-download-assets --assets-path ~/.robocasa/assets --no-macros -y
 export ROBOCASA_ASSETS_PATH=~/.robocasa/assets
 ```
```
Comment (QuanluZhang) [inline robots/robocasa/README.md]: what is robocasa/README.md used for? can we move it to the doc?

Comment (ye1457) [reply robots/robocasa/README.md]: Agreed. The separate README was intended as a GitHub entry point, but maintaining the same setup instructions twice is unnecessary.

In commit 5f2f1c4, I removed `robots/robocasa/README.md` and moved its unique runtime flow and troubleshooting details into both English and Chinese RoboCasa usage pages. The root README now links to the published usage guide; the test-guide link is updated in #159 as well. Both usage pages also document the installer recovery changes and describe the four representative environment checks, separately from the 340-cell evaluation protocol.

Rebased this PR onto current main (`3f6e247`), preserving the upstream Franka changes. Validation: both strict Sphinx builds, pre-commit, link checks and wheel build passed; fresh Python 3.10 unit tests report **253 passed, 3 skipped** (optional RLinf integration tests). The four real RTX 4090 environment/camera smoke checks in #159 passed without skips. This remains Draft pending the dependency releases and the existing release/accuracy gates.

File: robots/robocasa/robot_spec.py [hunk 2]
```diff
@@ -218,6 +218,11 @@ def _add_cli_args(parser: argparse.ArgumentParser, use_dashboard: bool) -> None:
         default=None,
         help="RLDX checkpoint path for locally spawned vla_server",
     )
+    parser.add_argument(
+        "--vla-support-revision",
```
Comment (QuanluZhang) [inline robots/robocasa/robot_spec.py]: Three names for one concept: `backbone_revision (RLDX)`, `--support-revision (vla_server)`, `--vla-support-revision (robot_spec)`. "Support" doesn't say support of what; --vla-backbone-revision would be self-describing and match upstream.

Comment (ye1457) [reply robots/robocasa/robot_spec.py]: Agreed. Updated in 4ba0267 to use the same backbone concept throughout:

- RPent CLI: `--vla-backbone-revision`
- VLA server CLI: `--backbone-revision`
- `RoboCasaVLAFacade` and the upstream RLDX loader: `backbone_revision`

Worker argument forwarding, external-server diagnostics, both usage translations and the opt-in inference test now use the new names. These options are still unpublished in this Draft, so I removed the old spellings rather than preserving extra aliases. The default remains `None`; the pinned revision and model-loading behavior are unchanged.

Validation: **255 CPU tests passed, 3 skipped** for the absent optional RLinf dependency; pre-commit and both strict Sphinx builds passed. Tests cover the public option parser, worker forwarding, server CLI parsing and facade-to-RLDX forwarding, with and without an explicit revision. Both CLI help commands also expose the new names. No GPU/policy evaluation was rerun for this naming-only change.

**Comment by ye1457** [comment]
Synchronized in bb96356 and rebased onto current main (`328645e`). The dependency PRs and #159 are merged. RoboCasa now uses the maintained `rpent-robocasa365` package; Target50 explicitly pins the public merged RoboCasa/RLDX commits alongside its existing Robosuite commit. The duplicate README remains removed, both root README languages link to the usage docs, and the four-case smoke guidance is no longer duplicated. Backbone option naming remains consistent with the earlier review request.

Current validation: **411 CPU tests passed (6 skipped)**, **4/4 real GPU environment/camera checks**, **25** merged RoboCasa dependency tests and **9** merged RLDX revision tests. Both strict docs builds, pre-commit, wheel builds and scans passed. The constrained source install resolves from an empty Python 3.10 environment/fresh metadata cache; this is a dry-run, not a full clean install.

Marked Ready for review as requested. The PR description explicitly retains the remaining full-download/fresh-runtime, final-source offline inference and six-cell accuracy checks; no new policy accuracy or complete reproduction claim is made.

File: tests/integration_tests/robots/robocasa/test_rldx_support_smoke.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
+# Copyright 2026 The RPent Authors.
```
Comment (wilburx813) [inline tests/integration_tests/robots/robocasa/test_rldx_support_smoke.py]: already have a similar test in tests/e2e_tests

File: robots/robocasa/eval/target50-constraints.txt [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
+# The companion override file freezes the simulator and RLDX Git revisions.
 mujoco==3.3.1
 numpy==1.26.4
 pydantic==2.13.5
 # RPent's API planner contract targets the instruction semantics introduced at 2.1.
 pydantic-ai-slim==2.1.0
 rlinf-rldx==1.0.1.post10
-rlinf-robocasa365==1.0.1
+rpent-robocasa365==1.0.1
 torch==2.7.0
```
Comment (wilburx813) [inline robots/robocasa/eval/target50-constraints.txt]: pin torch version seems werid

File: docs/source-zh/rst_source/usage/robocasa.rst
Comment (wilburx813) [inline docs/source-zh/rst_source/usage/robocasa.rst]: seems it‘s env-specific and hacky，I think we should let user select their own compatible torch/cuda version

File: robots/robocasa/robot_spec.py [hunk 5]
```diff
@@ -218,6 +218,11 @@ def _add_cli_args(parser: argparse.ArgumentParser, use_dashboard: bool) -> None:
         default=None,
         help="RLDX checkpoint path for locally spawned vla_server",
     )
+    parser.add_argument(
+        "--vla-backbone-revision",
+        default=None,
+        help="Hub revision for RLDX backbone config/tokenizer (local VLA only)",
+    )
```
Comment (wilburx813) [inline robots/robocasa/robot_spec.py]: remove this arg？User following setup doc don‘t need to specific manage a seperate vla-backbone-revision

File: tests/integration_tests/robots/robocasa/test_rldx_support_smoke.py (hunk 3 above)
Comment (ye1457) [reply tests/integration_tests/robots/robocasa/test_rldx_support_smoke.py]: Agreed. Removed the added `tests/integration_tests/robots/robocasa/test_rldx_support_smoke.py` in 7128094. Both usage translations now point to the existing `tests/e2e_tests/robocasa/test_components.py::test_rldx_component`, which already exercises the VLA worker, HTTP RPC, modality configuration and first inference. The documented node collects successfully. No second GPU fixture or test matrix was added. Full VLA inference was not rerun because the shared GPU has insufficient free capacity without interrupting existing workloads.

File: robots/robocasa/eval/target50-constraints.txt (hunk 4 above)
Comment (ye1457) [reply robots/robocasa/eval/target50-constraints.txt]: Removed the Torch and torchvision exact pins from `target50-constraints.txt` in 7128094. The manifest retains the previously used versions only under `reference_accelerator` with `enforced: false`; its constrained package map no longer includes them. Tests enforce that separation. A dependency dry-run with a user-selected Torch 2.8.0 / torchvision 0.23.0 / CUDA 12.8 pair resolved successfully, demonstrating that the previous 2.7.0 pin is no longer required. This is resolver validation, not an accuracy-equivalence claim.

File: docs/source-zh/rst_source/usage/robocasa.rst
Comment (ye1457) [reply docs/source-zh/rst_source/usage/robocasa.rst]: Agreed. In 7128094, removed the prescribed `--torch-backend=cu126`, duplicated mirror-specific installation command and fixed FlashAttention wheel from both languages. The setup guide now asks users to select a compatible CUDA-enabled Torch/torchvision pair for their GPU/driver/Python using the official PyTorch installation guide, then install RPent with the non-accelerator constraints. FlashAttention is optional, with a link to its compatibility guidance. Actual accelerator versions should be recorded with results and checked through the existing E2E component test; both strict documentation builds passed.

File: robots/robocasa/robot_spec.py (hunk 5 above)
Comment (ye1457) [reply robots/robocasa/robot_spec.py]: Removed both `--vla-backbone-revision` and the server's `--backbone-revision` in 7128094. `robot_spec.py` is now identical to main; users only follow the setup guide and provide the usual checkpoint path.

The RoboCasa VLA worker automatically supplies the guide's validated support snapshot to the upstream loader. This keeps actual tokenizer/config loading aligned with the documented download without an extra knob or editing cached `main` refs. It applies to backbone metadata only, not the finetuned weights. Local and separately started RPent VLA servers share this behavior. Unit tests check the automatic revision and absence of user options, and real offline config/processor/tokenizer loading passed. Full CPU suite: 408 passed, 6 skipped.

File: docs/source-zh/rst_source/usage/robocasa.rst [hunk 6]
```diff
@@ -22,42 +40,44 @@ RLDX-1 要求 Python ``3.10``。请创建独立环境，并通过 ``.[robocasa]`
...
-      --override robots/robocasa/eval/target50-overrides.txt \
-      --default-index https://mirrors.aliyun.com/pypi/simple \
-      --torch-backend=cu126
-
-.. note::
-
-   flash-attn 是可选的，缺少时 RLDX-1 会回退到 PyTorch SDPA。若要加快策略
-   前向，可安装预编译 wheel —— PyPI 上只有 sdist，直接
-   ``pip install flash-attn`` 会源码编译 10-20 分钟：
+      --override robots/robocasa/eval/target50-overrides.txt
```
Comment (wilburx813) [inline docs/source-zh/rst_source/usage/robocasa.rst]: remove this override txt？since specific commit of reproducing is mentioned in the doc，override the setup seems useless.

Comment (ye1457) [reply docs/source-zh/rst_source/usage/robocasa.rst]: Removed in `1b0d895`, thank you. `target50-overrides.txt`, the manifest field, `--override` setup options and the override-file contract are gone. Both usage guides now keep ordinary setup on the maintained `rpent` branches with compatibility constraints only. The already documented formal revisions are installed explicitly with a standard source-only `uv pip install --no-deps` step and `uv pip check`, so we do not silently describe a floating-branch install as a frozen one. No replacement override/helper was added.

After rebasing onto main `28bff49`: 460 CPU tests passed (3 skipped); pre-commit, both strict documentation builds, wheel and content scans passed. Resolver dry-runs passed for setup without overrides and the separate three-source install. No GPU policy run or frozen experiment was changed.

**Review by wilburx813 — APPROVED** [review]

File: pyproject.toml [hunk 7]
```diff
@@ -115,7 +115,7 @@ franka = [
 ]
 robocasa = [
     "rpent[rlinf]",
-    "rlinf-robocasa365",
+    "rpent-robocasa365 @ git+https://github.com/RLinf/robocasa.git@rpent",
```
Comment (qurakchin) [inline pyproject.toml:118]: use rlinf-robocasa365 is ok

File: tests/unit_tests/robots/robocasa/test_target50_protocol_contracts.py [hunk 8]
```diff
@@ -111,19 +112,31 @@ def test_target50_constraints_match_manifest_packages():
...
-        if line.strip() and not line.startswith("#")
-    }
-
-    assert overrides == {
-        f"robosuite @ git+https://github.com/RLinf/robosuite.git@{revision}"
+def test_target50_source_revisions_match_documented_install_commands():
+    dependencies = _manifest()["dependencies"]
+    sources = {
+        "robosuite": ("robosuite", "97cfbde4b68d8ec43dad20cf4747297866a6ca2e"),
+        "robocasa": ("rpent-robocasa365", "2692d8fc5fd86708a1b2028dcbce892ec418a9e7"),
```
Comment (qurakchin) [inline tests/unit_tests/robots/robocasa/test_target50_protocol_contracts.py]: use rpent branch, not the commit. robosuite / rldx are same

File: robots/robocasa/eval/target50.json [hunk 9]
```diff
@@ -9,21 +9,31 @@
...
-        "rlinf-robocasa365": "1.0.1",
-        "torch": "2.7.0",
-        "torchvision": "0.22.0",
+        "rpent-robocasa365": "1.0.1",
         "transformers": "4.57.6"
       }
     },
+    "robocasa": {
+      "repository": "https://github.com/RLinf/robocasa",
+      "commit": "2692d8fc5fd86708a1b2028dcbce892ec418a9e7"
```
Comment (qurakchin) [inline robots/robocasa/eval/target50.json]: for repos have the rpent branch, use the branch not the commit.

File: pyproject.toml (hunk 7 above)
Comment (ye1457) [reply pyproject.toml:118]: Checked the metadata on the maintained `rpent` branch: [setup.py](https://github.com/RLinf/robocasa/blob/2692d8fc5fd86708a1b2028dcbce892ec418a9e7/setup.py) declares `rpent-robocasa365`. I have retained that distribution name for now: changing only the requirement to `rlinf-robocasa365` would mismatch the package metadata, and switching to the other branch would omit the merged fixes. The docs now explain the actual name and warn against co-installing both distributions. Happy to follow a coordinated upstream rename once its metadata is updated. This naming point remains open; I am not treating it as accepted. Branch-policy update: 43899b0.

File: tests/unit_tests/robots/robocasa/test_target50_protocol_contracts.py (hunk 8 above)
Comment (ye1457) [reply tests/unit_tests/robots/robocasa/test_target50_protocol_contracts.py]: Updated in 43899b0. RoboCasa, Robosuite and RLDX now consistently use their `rpent` branches in the extras, Target50 manifest and contract tests. Removed the separate fixed-SHA installation step; both guides retain compatibility checks and ask users to record resolved commits with `uv pip freeze`. HF checkpoint, backbone-support and memory revisions are unchanged. Validation: 588 CPU tests passed (3 skipped), pre-commit and strict English/Chinese Sphinx builds passed. Fresh Python 3.10.12 environment installation has now also passed: 261 packages satisfy `uv pip check`; all three installed Git dependencies report requested revision `rpent`; RPent/RoboCasa CLI help, asset installer `--overwrite`, and RLDX/VLA imports pass. A second installation also succeeded. This used official/public package sources and verified download caches, not copied site-packages or local source patches. GPU inference and the six-cell gate were not run in this check.

File: robots/robocasa/eval/target50.json (hunk 9 above)
Comment (ye1457) [reply robots/robocasa/eval/target50.json]: Done in 43899b0: all three Git dependencies in `target50.json` now record `branch: "rpent"`, with tests checking the same branch requirements in the `robocasa` extra. Removed the obsolete commit-pinned source-install commands from both guides. The docs explicitly distinguish moving source branches (record each resolved commit) from the unchanged immutable HF resource snapshots. No task/seed, budget or scoring changes.

**Review by qurakchin — APPROVED** [review]
